### ISSUE TITLE
Out of memory error when attempting to import a 400 + page comic

### ISSUE THREAD

**Issue by bareheiny** [mcpierce/comixed#473]
[CMD ProblemFile.txt](https://github.com/mcpierce/comixed/files/3269339/CMD.ProblemFile.txt)

I run CX from a bat file, which outputs the CX log to log file.  

Generally, I don't expect any output in the CMD window, however when attempting to import a comic that has 475 pages a error is shown in the CMD window.

I have attached the CMD window output.  Unfortunately I did not keep the CX log file - however I can retry the import process to generate one.

This *seems* to be when the CX import window no longer updates

Note the reference to the hash in the text file was for use when the CX log was kept - that appears to be when the memory error was thrown.

**Comment by mcpierce** [comment]
I don't see an attachment file. Can you regenerate it and try attaching it again?

**Comment by bareheiny** [comment]
Odd - I can see it.

It's not showing as an attachement htough - just the blue "CMD ProblemFile.txt" as a hyperlink.

**Comment by bareheiny** [comment]
Seems to be resolved by increasing the max memory on java run.

Will likely need to create a memory issue bug or feature request at a later date.

**Comment by mcpierce** [comment]
@bareheiny Would you mind adding here what you did to increase memory so, if someone else hits the same issue, they can do the same?

I'm putting into the backlog to refactor how data is sent down to the frontend to reduce memory consumption. I think increasing heapspace is only going to push off the error for a time. The biggest memory hog is having the Comic load each of its Page objects when fetching the library. So that's where I'll focus when I address this issue.

**Comment by bareheiny** [comment]
This is the command I use to start CX, the bold part increases the maximum available mermory to 2GB (an arbitrary number, I just picked half of the physical memory in machine to see what happens).

java **-Xmx2g** -jar comixed-app.jar


Fully agree, increasing the memory just pushes the problem aside for a bit, but I was able to get 43k comics imported - before I increased the memory, it was topping out around 25k.  But loading the library still takes some time, and I wasn't able to start the import process again - although I likely didn't try to much, as I was focusing on finding the files that were causing errors.
